app/assets/javascripts/darkswarm/directives/loading.js.coffee

```diff
@@ -1,9 +0,0 @@
-Darkswarm.directive "loading", (Loading)->
-  # Triggers a screen-wide "loading" thing when Ajaxy stuff is happening
```

Comment:
I appreciate that a lot of work has gone into making this DRY, but I think sometimes it is possible to take that idea too far. In this case, it feels to me as though some code duplication would have been preferable to bringing the Loading, Navigation and RailsFlashLoader classes into the shared module. Going by my testing, the Loading overlay and the flash are not being used/rendered properly in the admin section anyway, so it feels like a muddying of the waters having this code here. What do you think @stveep?

Reply:
I can duplicate it if you prefer but I don't really see the problem with extracting the Navigation module; this could probably find other uses in the backend too. 

With Loading and FlashLoader, the problem is that these are required in the front end implementation so we still need to satisfy the requirement for them when the Stripe stuff is extracted to be shared with the backend. That was really the only reason for moving these. Do you think there's a better structure for how the dependencies are handled that we could use to avoid having to require these in the backend? As you say, it doesn't really work properly and it would be better to have the notifications/status updates more consistent with the rest of the admin interface - using statusMessage for example - but then we will have the opposite problem in the front end. I could add some logic for this to change how it's handled depending on whether we are in checkout or admin, but would still end up needing to include all the different methods.

Reply:
Thanks @stveep 

>this could probably find other uses in the backend too

To me, adding the `Navigation` service to the admin section on the off-chance that we might use it is a case of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). Also, IMHO, the `Navigation` service is not actually all that useful. It's just a very thin wrapper for the $location service that ships with AngularJS. For example, I wrote the `tabsetCtrl` directive which uses paths and doesn't depends on the `Navigation` service.

My preference would be that we duplicate the relevant parts of the `stripeElements` directive and `StripeElements` service from the frontend to admin, and rewrite them to use the equivalent admin services as required. Then we can keep the `Navigation`, `Loading` and `RailsFlashLoader` services in the frontend, thus removing the need for any shared modules.

I think the only real alternative is to do more work to refactor the Stripe directive and service so that they don't directly depend on any other services (using `$injector` you can check if services are present before using them), so that they can be cleanly used in either context. This still feels pretty messy to me, and will complicate the Stripe logic with lots of conditionals, but we can investigate this more if you'd prefer to go down that path.